front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -75,23 +80,20 @@ const RangeEditionLeftPanel = () => {
   };
 
   const toggleSpeedRestriction = () => {
-    if (speedRestriction) resetSpeedRestrictionSelections();
-    setSpeedRestriction(!speedRestriction);
-    const selectiontype = speedRestriction ? 'idle' : 'selectSwitch';
+    if (isSpeedRestriction) resetSpeedRestrictionSelections();
+    const selectiontype = isSpeedRestriction ? 'idle' : 'selectSwitch';
     const newEntity = cloneDeep(entity) as SpeedSectionEntity;
     if (newEntity.properties.extensions) {
       newEntity.properties.extensions = undefined;
     }
```

Comment:
I think the test is not really needed, you can directly set the extensions to undefined
```suggestion
    newEntity.properties.extensions = undefined;
    
```

Reply:
done